### PR TITLE
Fixes #2047: PHPMessDetectorBearTest: Closed opened test files properly

### LINKED ISSUE #2047: PHPMessDetectorBearTest: Close opened test files properly
py.test gives warnings on https://circleci.com/gh/coala/coala-bears/7126 like
```
tests/php/PHPMessDetectorBearTest.py::PHPMessDetectorBearTest::test_cleancode_violation
  /home/ubuntu/coala-bears/tests/php/PHPMessDetectorBearTest.py:19: ResourceWarning: unclosed file <_io.TextIOWrapper name='/home/ubuntu/coala-bears/tests/php/phpmessdetector_test_files/cleancode_violation.php' mode='r' encoding='UTF-8'>
    return open(get_testfile_path(name)).readlines()
```
To not rely on ref-counting for object destruction (which isn't anyway an official feature in Python to allow custom destructors), we should close the file manually by using a with-context:
```
with open(get_testfile_path(name)) as fl:
    return fl.readlines()
```

### PR DESCRIPTION
**Pull request by vaibhavsingh97**
Fixes #2047 

**For short term contributors:** we understand that getting your commits well
defined like we require is a hard task and takes some learning. If you
look to help without wanting to contribute long term there's no need
for you to learn this. Just drop us a message and we'll take care of brushing
up your stuff for merge!

### Checklist

- [x] I read the [commit guidelines](http://coala.io/commit) and I've followed
      them.
- [x] I ran coala over my code locally. (*All commits have to pass
      individually.* It is not sufficient to have "fixup commits" on your PR,
      our bot will still report the issues for the previous commit.) You will
      likely receive a lot of bot comments and build failures if coala does not
      pass on every single commit!

After you submit your pull request, **DO NOT click the 'Update Branch' button.**
When asked for a rebase, consult [coala.io/rebase](https://coala.io/rebase)
instead.

Please consider helping us by reviewing other peoples pull requests as well:

- pick up any PR at <https://coala.io/review>
- review it (check <https://coala.io/reviewing> for more info)
- if you are sure that it needs work, use `cobot mark wip <URL>` to get it out
  of the review queue.

The more you review, the more your score will grow at coala.io and we will
review your PRs faster!

### CODE REVIEW

**Comment by khanchi97** [comment]
@vaibhavsingh97 fix issues given by gitmate-bot

**Comment by khanchi97** [comment]
you can take help from this http://api.coala.io/en/latest/Developers/Writing_Good_Commits.html

**Comment by adtac** [comment]
unack https://github.com/coala/coala-bears/pull/2082/commits/36906d107b702727f1f50c21dcd4b49a3f5051a1

**Comment by vaibhavsingh97** [comment]
@prakashdanish @khanchi97 I will edit my Commit but why circle Ci is failing? I tried to open but it's not loading the error. Can anyone help?

**Comment by khanchi97** [comment]
@vaibhavsingh97 if you download the full output as a file then you will see it is failing just because of 
GitCommitBear means if you fix your commit message then CicleCI would not fail

**Comment by RaiVaibhav** [comment]
@vaibhavsingh97  Use 
```
PHPMessDetectorBearTest: Close test files properly

Closes https://github.com/coala/coala-bears/issues/2047
```

**Comment by vaibhavsingh97** [comment]
@RaiVaibhav Thanks 😄

**Comment by RaiVaibhav** [comment]
@vaibhavsingh97  new-comer guide is best take a look at it

**Comment by RaiVaibhav** [comment]
@vaibhavsingh97 do the changes and rebuild it

**Comment by khanchi97** [comment]
@prnvdixit  afaik i think it works, if in your commit you just use issue no. followed by # i.e `#2047`.isn't it? :thinking:

**Comment by RaiVaibhav** [comment]
The problem is in [line-302](https://github.com/coala/coala-bears/blob/6905edcf6fc93bdff46ec6ba69fd8cd09232853a/bears/vcs/git/GitCommitBear.py#L302)
```
Traceback (most recent call last):
  File "/home/ubuntu/.pyenv/versions/3.5.1/lib/python3.5/site-packages/coalib/bears/Bear.py", line 308, in execute
    return [] if result is None else list(result)
  File "/home/ubuntu/.pyenv/versions/3.5.1/lib/python3.5/site-packages/bears/vcs/git/GitCommitBear.py", line 134, in run
    **self.get_issue_checks_metadata().filter_parameters(kwargs))
  File "/home/ubuntu/.pyenv/versions/3.5.1/lib/python3.5/site-packages/bears/vcs/git/GitCommitBear.py", line 302, in check_issue_reference
    body = body.splitlines()[-1]
IndexError: list index out of range
```
Try to add a blank line after your `Closes`
I am actually confuse and yes use the issue url link

**Comment by RaiVaibhav** [comment]
@vaibhavsingh97 does space after a `Closes` fix the problem or using `Url`??

**Comment by vaibhavsingh97** [comment]
@RaiVaibhav I added `url`

**Comment by prnvdixit** [comment]
ack bd09c5249212e8b0fd8c29147bb1d95263fd0c3b

**Comment by RaiVaibhav** [comment]
@prnvdixit you are right i checked  that on my PC

**Comment by khanchi97** [comment]
@prnvdixit oh yeah, I didn't know that.thank for sharing this. :+1:

**Comment by vaibhavsingh97** [comment]
Can anyone please merge this 😄

File: tests/php/PHPMessDetectorBearTest.py [hunk 1]
```diff
@@ -16,7 +16,8 @@ def get_testfile_path(name):
 
 
 def load_testfile(name):
-    return open(get_testfile_path(name)).readlines()
+    with open(get_testfile_path(name)) as f1:
```
Comment (shreyans800755) [inline tests/php/PHPMessDetectorBearTest.py]: Can you please rename `f1` to something meaningful ? May be `file` ?

**Comment by shreyans800755** [comment]
unack bd09c52

File: tests/php/PHPMessDetectorBearTest.py (hunk 1 above)
Comment (vaibhavsingh97) [reply tests/php/PHPMessDetectorBearTest.py]: @shreyans800755 sure 👍

**Comment by vaibhavsingh97** [comment]
@shreyans800755 Done 👍 Please review 😄

**Comment by shreyans800755** [comment]
@gitmate-bot rebase

**Comment by shreyans800755** [comment]
ack 8eae719

**Comment by shreyans800755** [comment]
@vaibhavsingh97 PR looks good. Please ask any of the maintainer to merge it.

**Comment by jayvdb** [comment]
@rultor merge

**Pull request merged**